frontend/src/component/demo/DemoTopics/DemoTopics.tsx

```diff
@@ -0,0 +1,210 @@
+    borderTopLeftRadius: theme.shape.borderRadiusLarge,
+    borderTopRightRadius: theme.shape.borderRadiusLarge,
+}));
+
+const StyledExpandMoreIcon = styled(ExpandMore)(({ theme }) => ({
+    color: theme.palette.primary.contrastText,
+}));
+
+const StyledTitle = styled('div')({
+    display: 'flex',
+    alignItems: 'center',
+});
+
+const StyledSubtitle = styled(Typography)(({ theme }) => ({
+    fontSize: theme.fontSizes.smallerBody,
+    marginTop: theme.spacing(0.5),
+    marginBottom: theme.spacing(0.5),
+}));
+
+const StyledProgress = styled('div')(({ theme }) => ({
+    width: '100%',
+    display: 'flex',
+    alignItems: 'center',
+    gap: theme.spacing(1.5),
+}));
+
+const StyledLinearProgress = styled(LinearProgress)(({ theme }) => ({
+    width: '100%',
+    height: theme.spacing(1),
+    borderRadius: theme.shape.borderRadius,
+    [`&.${linearProgressClasses.colorPrimary}`]: {
+        backgroundColor: theme.palette.primary.dark,
+    },
+    [`& .${linearProgressClasses.bar}`]: {
+        borderRadius: theme.shape.borderRadius,
+        backgroundColor: theme.palette.primary.contrastText,
+    },
+}));
+
+const StyledStep = styled('li', {
+    shouldForwardProp: prop => prop !== 'selected' && prop !== 'completed',
+})<{ selected?: boolean; completed?: boolean }>(
+    ({ theme, selected, completed }) => ({
+        padding: theme.spacing(1),
+        cursor: 'pointer',
+        display: 'flex',
+        alignItems: 'center',
+        marginTop: theme.spacing(1),
+        borderRadius: theme.shape.borderRadius,
+        gap: theme.spacing(1),
+        backgroundColor: theme.palette.background.elevation2,
+        ...(selected && {
+            backgroundColor: theme.palette.secondary.light,
+            fontWeight: theme.typography.fontWeightBold,
+            border: `1px solid ${theme.palette.primary.main}`,
+        }),
+        ...(completed && {
+            backgroundColor: theme.palette.background.elevation1,
+            textDecoration: 'line-through',
+        }),
+    })
+);
+
+const StyledCheckCircle = styled(CheckCircle)(({ theme }) => ({
+    color: theme.palette.primary.main,
+    fontSize: theme.fontSizes.bodySize,
+}));
+
+const StyledCircleOutlined = styled(CircleOutlined)(({ theme }) => ({
+    color: theme.palette.neutral.main,
+    fontSize: theme.fontSizes.bodySize,
+}));
+
+const StyledStepIcon = styled(ExpandMore)(({ theme }) => ({
+    transform: 'rotate(-90deg)',
+    fontSize: theme.fontSizes.bodySize,
+}));
+
+const StyledButton = styled(Button)(({ theme }) => ({
+    width: '100%',
+    marginTop: theme.spacing(2),
+    '&&&': {
+        fontSize: theme.fontSizes.smallBody,
+    },
+}));
+
+interface IDemoTopicsProps {
+    expanded: boolean;
+    setExpanded: React.Dispatch<React.SetStateAction<boolean>>;
+    steps: number[];
+    currentTopic: number;
+    setCurrentTopic: (topic: number) => void;
+    topics: ITutorialTopic[];
+}
+
+export const DemoTopics = ({
+    expanded,
```

Comment:
I'm not sure how much this component will be reused in different context but usually my recommendation is to avoid passing props and use slots (ReadNode children) instead. Or composition over configuration as it's called. https://artursabirov.blog/composition-over-configuration-in-react/. E.g. in HTML you're not passing <select options=[props]> but <select><option><option></select>. This make for much more extensible APIs.

Reply:
Hi @kwasniew - From reading your comment I got the impression that maybe it was left unfinished. 
I don't foresee this component to be reused, at least not at this stage, and there's little drill down happening. The component currently exists only to abstract way some of the complexity and isolated logic of displaying the topics on the bottom left corner. Going for a composition over configuration refactor here may leak things to the parent component that it doesn't need / should care about.

Given this specific scenario, could you please share an example of what you have in mind? Maybe it would help me visualize it and better understand the pros vs cons here.

Reply:
GH actions didn't display my full comment :)

I'd expect this type of API, but again if it's not reused then I'm ok with your solution:
```
<DemoTopics>
   <Topic selected/>
  <Topic/>
  <Topic/>
</DemoTopics>
```
BTW, The point of composition is not only to avoid prop drilling but to have extensible UI. E.g. If you wan to have enhanced topic you can do it without touching DemoTopics every single time.

```
<DemoTopics>
   <WithPicture><Topic selected/></WithPicture>
  <Topic/>
  <Topic/>
</DemoTopics>
```